Indicators/T3MovingAverage.cs

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the T3 Moving Average (T3). 
+    /// The T3 Moving Average is calculated with the following formula:
+    /// EMA1(x, Period) = EMA(x, Period)
+    /// EMA2(x, Period) = EMA(EMA1(x, Period),Period)
+    /// GD(x, Period, volumeFactor) = (EMA1(x, Period)*(1+volumeFactor)) - (EMA2(x, Period)* volumeFactor)
+    /// T3 = GD(GD(GD(t, Period, volumeFactor), Period, volumeFactor), Period, volumeFactor);
+    /// </summary>
+    public class T3MovingAverage : IndicatorBase<IndicatorDataPoint>
+    {
+        private readonly int _period;
+        private readonly DoubleExponentialMovingAverage _gd1;
+        private readonly DoubleExponentialMovingAverage _gd2;
+        private readonly DoubleExponentialMovingAverage _gd3;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T3MovingAverage"/> class using the specified name and period.
+        /// </summary> 
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of the T3MovingAverage</param>
+        /// <param name="volumeFactor">The volume factor of the T3MovingAverage</param>
+        public T3MovingAverage(string name, int period, int volumeFactor) 
+            : base(name)
+        {
+            _period = period;
+            _gd1 = new DoubleExponentialMovingAverage(name + "_1", period, volumeFactor);
+            _gd2 = new DoubleExponentialMovingAverage(name + "_2", period, volumeFactor);
+            _gd3 = new DoubleExponentialMovingAverage(name + "_3", period, volumeFactor);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T3MovingAverage"/> class using the specified period.
+        /// </summary> 
+        /// <param name="period">The period of the T3MovingAverage</param>
+        /// <param name="volumeFactor">The volume factor of the T3MovingAverage</param>
+        public T3MovingAverage(int period, int volumeFactor)
+            : this(string.Format("T3({0},{1})", period, volumeFactor), period, volumeFactor)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady
+        {
+            get { return Samples > 6 * (_period - 1); }
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            _gd1.Update(input);
+
+            if (Samples <= 2 * (_period - 1))
```

Comment:
Is this really asking `if (_gd1.IsReady) { }` ? Same for the below `if` checks against the `Samples` property